pkg/state/state.go

```diff
@@ -83,6 +84,30 @@ type ReleaseSetSpec struct {
 	LockFile string `yaml:"lockFilePath,omitempty"`
 }
 
+// helmStateAlias is helm state alias
+type helmStateAlias HelmState
+
+func (hs HelmState) MarshalYAML() (interface{}, error) {
+	return helmStateAlias(hs), nil
+}
+
+func (hs *HelmState) UnmarshalYAML(value *yaml.Node) error {
+	helmStateInfo := make(map[string]interface{})
+	if err := value.DecodeWithOptions(&helmStateInfo, yaml.DecodeOptions{KnownFields: true}); err != nil {
```

Comment:
```suggestion
	if err := value.Decode(&helmStateInfo); err != nil {
```

Can we probably use the vanilla Decode function because KnownFields might make sense only when you decode into a struct?

Reply:
We need this feature, and with the addition of `UnmarshalYAML`, the parsing process is under our own control.

Reply:
You'd need it when you're decoding it into a struct in `return value.DecodeWithOptions((*helmStateAlias)(hs), yaml.DecodeOptions{KnownFields: true})`, yeah.
However, you don't need KnownFields as the target isn't a struct for this line, right?

Reply:
@mumoshu When a struct defines an `UnmarshalYAML` method, its normal parsing flow will call it, which means our normal parsing logic will change, and `KnownFields` is needed to keep it compatible.

Reply:
That makes sense! Thanks a lot for clarifying ☺️